### PR TITLE
fuzz: Adds a structured fuzzer

### PR DESCRIPTION
**Pull request by silvergasp**
This is a precursor to full roundtrip property testing as mentioned in https://github.com/tafia/quick-xml/pull/554#issuecomment-1435310267.

### CODE REVIEW

**Comment by Mingun** [comment]
How we can check, that new fuzz target is executed? I see no obvious indicators in the [results](https://github.com/tafia/quick-xml/actions/runs/4246355859/jobs/7383054689). `fuzz_target_1` is found [in logs](https://pipelines.actions.githubusercontent.com/serviceHosts/a318cdfd-6c50-401d-9888-66d4456ae54d/_apis/pipelines/1/runs/689/signedlogcontent/2?urlExpires=2023-02-23T05%3A01%3A27.8352727Z&urlSigningMethod=HMACV1&urlSignature=%2F5%2FuAUXJPuMB9PMFCHYvlcBY%2FUM4KVaO%2Bd6sdSLtLUk%3D) page 27 times (7 times in **Run Fuzzers** step), and `structured_roundtrip` is only one (and only as mention of file name that is changed in PR).

**Comment by silvergasp** [comment]
Good catch! It currently isn't running, in the CI. I'll open a PR on google/oss-fuzz so that new fuzz tests, are automatically detected.

**Comment by silvergasp** [comment]
This PR upstream should include this new fuzz harness and should autodetect all future fuzz harnesses 
https://github.com/google/oss-fuzz/pull/9795.

**Review by Mingun — APPROVED** [review]
Ok. Then I want to wait a bit until https://github.com/google/oss-fuzz/pull/9795 is merged just to rerun the checks and to be sure that's all is working (unfortunately, I cannot run checks locally by various reasons :(). But if merging of upstream would delay more than a week I'll merge it anyway.

Oh, just one note -- can you move `Arbitrary` implementations after `Deref` implementations?

**Comment by Mingun** [comment]
The documentation for [size_hint](https://docs.rs/arbitrary/latest/arbitrary/trait.Arbitrary.html#method.size_hint) suggest to implement it when implementing trait manually. Because events definitely used in collections of events it is better to override default behavior.

**Comment by silvergasp** [comment]
> The documentation for [size_hint](https://docs.rs/arbitrary/latest/arbitrary/trait.Arbitrary.html#method.size_hint) suggest to implement it when implementing trait manually. Because events definitely used in collections of events it is better to override default behavior.

True, I've implemented these now.

File: src/events/mod.rs [hunk 1]
```diff
@@ -619,6 +650,16 @@ pub struct BytesText<'a> {
     decoder: Decoder,
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for BytesText<'a> {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let s = <&str>::arbitrary(u)?;
+        if s.is_empty() || !s.chars().all(char::is_alphanumeric) {
```
[the remark this answers is not in the document]
Comment (silvergasp) [reply src/events/mod.rs]: Done

**Comment by silvergasp** [comment]
https://github.com/google/oss-fuzz/pull/9795 has been merged now. I'm just going to close/reopen this PR to re-trigger the CI to double check that CIfuzz captures the new fuzzer.

**Comment by silvergasp** [comment]
So the new fuzzer is now running in CI as well https://github.com/tafia/quick-xml/pull/560/checks#step:5:30

**Review by Mingun — APPROVED** [review]

**Pull request merged**